examples/esphttpd/cgi-test.c

```diff
@@ -0,0 +1,90 @@
+/*
+Cgi routines as used by the tests in the html/test subdirectory.
+*/
+
+/*
+ * ----------------------------------------------------------------------------
+ * "THE BEER-WARE LICENSE" (Revision 42):
+ * Jeroen Domburg <jeroen@spritesmods.com> wrote this file. As long as you retain 
+ * this notice you can do whatever you want with this stuff. If we meet some day, 
+ * and you think this stuff is worth it, you can buy me a beer in return. 
+ * ----------------------------------------------------------------------------
+ */
+
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <espressif/esp_common.h>
+
+#include "cgi-test.h"
+
+typedef struct {
+	int len;
+	int sendPos;
+} TestbedState;
+
+
+int ICACHE_FLASH_ATTR cgiTestbed(HttpdConnData *connData) {
+	char buff[1024];
```

Comment:
Looks very interesting. One nit the ICACHE_FLASH_ATTR might not be needed for esp-open-rtos as functions are linked to the flash by default.

Any chance that you could rework https://github.com/nochkin/libesphttpd so that the master branch follows the upstream, and create a separate branch for the esp-open-rtos modifications. That would make it much easier to follow upstream changes and to see what has changed.

Reply:
I have #define to null it. It's in platform.h file. This is done in hopes that I will push my changes to upstream some day.
It's currently forked from the master if that's what you mean.

Reply:
Yes, it needs zlib to build linux binaries for compression of static html/css/js files.
Can I add the zlib package or the maintainer has to do it?